neo/Network/P2P/LocalNode.cs

```diff
@@ -174,6 +223,60 @@ private void OnSendDirectly(IInventory inventory)
             Connections.Tell(inventory);
         }
 
+        protected override void TcpDisconnect(DisconnectReason reason)
+        {
+            var disconnectMessage = CreateDisconnectMessage(reason);
+            var command = Tcp.Write.Create(ByteString.FromBytes(disconnectMessage.ToArray()));
+
+            Sender.Tell(new Tcp.Register(ActorRefs.NoSender));
+            Sender.Ask(command).ContinueWith(t => Sender.Tell(Tcp.Abort.Instance));
+        }
+
+        protected override void WsDisconnect(WebSocket ws, DisconnectReason reason)
+        {
+            var disconnectMessage = CreateDisconnectMessage(reason);
+            ArraySegment<byte> segment = new ArraySegment<byte>(disconnectMessage.ToArray());
+
+            ws.SendAsync(segment, WebSocketMessageType.Binary, true, CancellationToken.None).PipeTo(Self,
+                failure: ex => new Tcp.ErrorClosed(ex.Message));
+            ws.CloseAsync(WebSocketCloseStatus.NormalClosure, "close ws", CancellationToken.None);
+        }
+
+        private Message CreateDisconnectMessage(DisconnectReason reason)
+        {
+            byte[] data;
+            switch (reason)
+            {
+                case DisconnectReason.MaxConnectionReached:
+                case DisconnectReason.MaxConnectionPerAddressReached:
+                    data = GetRandomConnectedPeers(MinDesiredConnections).ToByteArray();
```

Comment:
`MinDesiredConnections` may be different on different nodes. I think 10 is okay.

Reply:
ok, reset